### DISCUSSION TITLE
Source code of the sedimentation of 50 particles in a closed container example

### DISCUSSION THREAD

**Discussion by dfcaporale** [chaos-polymtl/lethe#622] [General]
Hi! I am wondering if there is a chance to publish the source code of [this example from YouTube](https://www.youtube.com/watch?v=lVqnZR4tybM&lc=UgxRlvN8vbicyH70Jk54AaABAg.9gDc19R69Ae9hQjjNpo2Dq) to reproduce it locally. Your simulations look amazing.

Thanks in advance, and congratulations on your code!

**Comment by Luckabarbeau** [comment]
Hi @dfcaporale,

I am Lucka Barbeau, the main developer of the resolved CFD-DEM solver. Sorry for the delay in responses, I was unavailable last week.

First, I would like the thank you for your interest in the resolved CFD-DEM code. We would be happy to share the .prm file associated with this case. Sadly the code has significantly evolved since this simulation was carried out, and the .prm file used for this case would not work with the current version. I work on very similar cases at the moment. If you are only interested in the sedimentation of multiple particles in general, I can send the .prm file I am working on at the moment. If you want this specific case for reproduction purposes, I will take the time necessary to update the .prm file of the case in the video to the current version.

Thank you again for your interest in our code. It is always an excellent way to start the week when I get questions like that.

Lucka

**Comment by dfcaporale** [comment]
Hi @Luckabarbeau,

Thanks for your answer. There is no hurry at all.

My interest is in the case of a bunch of solid particles moving immersed in a cyclic pipe flow. Since I was impressed by your simulations, I thought it could be done by modifying your YouTube example to reproduce my case study. So, if you can suggest me another already posted example that could fit my interest, that would be awesome.

Again, thank you so much for your answer and congratulations on your code and excitement about the subject!

Daniel

**Comment by Luckabarbeau** [comment]
Hi @dfcaporale,

Here you will find a simplified version of the case I am currently working on. This case is the sedimentation of 64 particles initially structured in a 4 by 4 by 4 arrangement in a rectangular container. This case should run on a machine with at least 16g of ram. This case has not been made into an example at the moment, but if you find it pertinent relative to more sampler but similar example ( e.g [Sedimentation of one particle](https://lethe-cfd.github.io/lethe/examples/resolved-cfd-dem/sedimentation-1-particle/sedimentation-1-particle.html))  we could write a dedicated example for this case. 

Here are the two files needed to run this case.
[sedimentation_4_4_4_case.zip](https://github.com/lethe-cfd/lethe/files/9911046/sedimentation_4_4_4_case.zip)

To run this case, use the gls_sharp_navier_stokes_3d solver and put the two files joined with this message in the same directory.

Note: I don't know if it would be required in your case or not, but at the moment, we don't support the implementation of periodic boundaries for the particles in resolved CFD-DEM.

Don't hesitate if you have any questions related to this solver. 

Lucka

**Reply by dfcaporale** [reply]
I'm so happy to receive such an answer! 
I'll try it in the following days and get back to you.
Although I'll need periodic BCs for the study case I'm tackling, I'll still consider other options in the meantime, while periodic BCs are not supported.

Once again: thank you so much for your help!
